### ISSUE TITLE
About Resuming Training

### ISSUE THREAD

**Issue by SmallLeishen** [Intellindust-AI-Lab/DEIMv2#40]
When use the DEIMv2 model with the version of "M" to train, I found that when I resumed training from last.pth which is not coming from the folder of "eval"(not latest.pth), If I'm currently in the augmentation phase, I can resume training normally. However, if I'm currently in the no augmentation stage, I can't resume training from last.pth, but instead resume training from an earlier checkpoint. Is this because I made the wrong choice? Do I need to make any special checkpoint choices to resume training in the second phase?
The resuming training code is as follows:
export CUDA_VISIBLE_DEVICES=0,5
export OMP_NUM_THREADS=1
torchrun --standalone --nnodes=1 --nproc_per_node=2 \
  --master_port=29511 \
  train.py \
  -c configs/deimv2/deimv2_dinov3_m_aitod2.yml \
  --seed 0 \
  -r /jsj_ljx/ldw/DEIMv2/output/deimv2_dinov3_m_aitod2_2/last.pth \
  --output-dir /jsj_ljx/ldw/DEIMv2/output/deimv2_dinov3_m_aitod2_2 \

**Comment by ShihuaHuang95** [comment]
@SmallLeishen Thank you very much for your interest in and attention to our work.

Regarding the phenomenon you mentioned: starting from RT-DETR and D-FINE, the training process has gradually evolved into two stages.

Stage 1 uses strong data augmentation, including multi-scale training.

Stage 2 disables all strong augmentations.

Especially beginning with D-FINE, Stage 2 also enables an EMA-decay search. We found this search to be effective, so we kept it in the pipeline. Under this design, once the model enters the EMA-decay search stage, it will only save the best Stage-2 model. The last-epoch model you see is the one from just before entering Stage 2.
